### ISSUE TITLE
(no library called "ssl" found)

### ISSUE THREAD

**Issue by fnkap** [LonamiWebs/Telethon#1200]
Hello guys.

I've got this error

```
2019-06-06 19:29:27,149 - telethon.crypto.libssl - INFO - Failed to load None: <class 'OSError'> (no library called "ssl" found)
2019-06-06 19:29:27,150 - telethon.crypto.aes - INFO - cryptg module not installed and libssl not found, falling back to (slower) Python encryption
```

My script is working fine, but it's slow making requests (as specified in second line)
I didn't install or uninstall anything and it worked until this morning.

Python 3.6.8, latest Telethon, Windows 10.

Any idea?

Already tried to go here

https://slproweb.com/products/Win32OpenSSL.html

And download Win64 OpenSSL v1.1.1c Light and a ton of other versions, so i don't think that's the problem.

Thank you for your replies.

**Comment by Lonami** [comment]
> and it worked until this morning

Do you mean it used to find `libssl` but now it doesn't anymore? In the mean time, you should `pip install cryptg` like the error suggests.

**Comment by fnkap** [comment]
> > and it worked until this morning
> 
> Do you mean it used to find `libssl` but now it doesn't anymore? In the mean time, you should `pip install cryptg` like the error suggests.

If I install cryptg the script uses it, but it's slow like before.
Until yesterday, it worked very fast, so it used libssl I suppose.
Also, I didn't have any warning until  yesterday.

**Comment by Lonami** [comment]
> but it's slow like before

That's highly unlikely, since `cryptg` does the work in C… There must be some improvement.

**Comment by fnkap** [comment]
Update: pushed my project to Heroku, with and without cryptg, it's blazing fast.

When I run it locally, with or without cryptg, it's really really slow. Like, 100x times slower. What it could be?

**Comment by Lonami** [comment]
Network speed, and if it's blazing fast in Heroku, it probably find `libssl` there. It's very very strange that you have `cryptg` installed locally but it works slow, are you sure it's installed?

**Comment by fnkap** [comment]
Problem is fixed when I remove an imported script. Guess it's something there which interferes. I will investigate more. Thank you for your time.

**Comment by Lonami** [comment]
Do you have a file called `C:\Windows\System32\libssl-1_1-x64.dll` (or similar)? What does:
```python
import ctypes.util
lib = ctypes.util.find_library('libssl-1_1-x64.dll')
print(lib)
ssl = ctypes.cdll.LoadLibrary(lib or r'C:\Windows\System32\libssl-1_1-x64.dll')
print(ssl)
```

Show?

**Comment by fnkap** [comment]
I think it was a problem with django. I imported django two times from two different scripts, and that caused the slowness, but only on windows.

I'm going to close the issue, because it's not relevand to Telethon. Thank you.

**Comment by Lonami** [comment]
Well... even if Django was the issue could you [run the code I wrote](https://github.com/LonamiWebs/Telethon/issues/1200#issuecomment-499822667)?

**Comment by fnkap** [comment]
Sure. This was the result:

```
Traceback (most recent call last):
  File "C:/Users/malef/Desktop/my_test.py", line 5, in <module>
    ssl = ctypes.cdll.LoadLibrary(lib or r'C:\Windows\System32\libssl-1_1-x64.dll')
  File "C:\Program Files\Python36\lib\ctypes\__init__.py", line 426, in LoadLibrary
    return self._dlltype(name)
  File "C:\Program Files\Python36\lib\ctypes\__init__.py", line 348, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: [WinError 127] Impossibile trovare la procedura specificata
```

**Comment by andydoc** [comment]
Shame the OP did not continue this - I did as you asked, as I am experiencing the same issue. 
The dll was not in System32, but was in my sublime text folder, from where I copied it. 
So the first print showed `C:\Program Files\Sublime Text\libssl-1_1-x64.dll`, the second showed `<CDLL 'C:\Program Files\Sublime Text\libssl-1_1-x64.dll, handle 7ffd6b9b0000 at 0x6403115c10>`. 
After copying the dll the printouts were no different. Running `regsvr32 C:\Windows\System32\libssl-1_1-x64.dll` produced a regsvr32 error regarding a failure to find an entry point DllRegisterServer... (see [here](https://photos.app.goo.gl/rWVDzyxihsoEVyMJ9))
